shared/haml/user_header.haml

```diff
@@ -37,6 +37,9 @@
   // Use function closure to avoid cluttering global namespace.
   (function() {
     var cookieKey = '#{'_shortName' + (rack_env?(:production) ? '' : "_#{rack_env}")}';
+    // Because we use rack_env in our cookie key, the client has no way of knowing
+    // what key to look for. Stash it on window so that clients can access this.
+    window.userNameCookieKey = cookieKey;
```

Comment:
Don't really like this, but not sure how I can do better. One option would be to instead pass `rack_env` and let client generate `cookieKey` from this

Reply:
I don't mind this actually - and I think it's strictly better than duplicating the logic for constructing the key from the environment on the client.  As a cross-cutting concern and (I think) strictly a constant within its `rack_env` I don't think this pollutes the global namespace too much.